tests/unittests/tests-hash_string/tests-hash_string.c

```diff
@@ -6,21 +6,28 @@
  * directory for more details.
  */
 
+#include <string.h>
 #include "tests-hash_string.h"
 #include "hash_string.h"
 
 static void test_hash_value(void)
 {
-    TEST_ASSERT_EQUAL_INT(177621, hash_string("0"));
-    TEST_ASSERT_EQUAL_INT(6555485335635392041, hash_string("0123456789abcde-"
-                                                           "0123456789abcde-0123"
-                                                           "456789abcde-01234567"
-                                                           "89abcde-"));
-    TEST_ASSERT_EQUAL_INT(210674770391, hash_string("Frank jagt im komplett"
-                                                    " verwahrlosten Taxi q"
-                                                    "uer durch Bayern"));
-    TEST_ASSERT_EQUAL_INT(249811328611268156, hash_string("12A5B6cd_"));
-    TEST_ASSERT_EQUAL_INT(5381, hash_string(""));
+    unsigned char string[100];
+
+    memcpy(string,"0",2);
+    TEST_ASSERT_EQUAL_INT(177621, hash_string(string));
+
+    memcpy(string,"0123456789abcde-0123456789abcde-0123456789abcde-0123456789abcde-",65);
```

Comment:
Why don't you just cast the strings?

```diff
-    TEST_ASSERT_EQUAL_INT(177621, hash_string("0"));
-    TEST_ASSERT_EQUAL_INT(6555485335635392041, hash_string("0123456789abcde-"
-                                                           "0123456789abcde-0123"
-                                                           "456789abcde-01234567"
-                                                           "89abcde-"));
-    TEST_ASSERT_EQUAL_INT(210674770391, hash_string("Frank jagt im komplett"
-                                                    " verwahrlosten Taxi q"
-                                                    "uer durch Bayern"));
-    TEST_ASSERT_EQUAL_INT(249811328611268156, hash_string("12A5B6cd_"));
-    TEST_ASSERT_EQUAL_INT(5381, hash_string(""));
+    TEST_ASSERT_EQUAL_INT(177621, hash_string((unsigned char *)"0"));
+    TEST_ASSERT_EQUAL_INT(6555485335635392041, hash_string((unsigned char *)"0123456789abcde-"
+                                                           "0123456789abcde-0123"
+                                                           "456789abcde-01234567"
+                                                           "89abcde-"));
+    TEST_ASSERT_EQUAL_INT(210674770391, hash_string((unsigned char *)"Frank jagt im komplett"
+                                                    " verwahrlosten Taxi q"
+                                                    "uer durch Bayern"));
+    TEST_ASSERT_EQUAL_INT(249811328611268156, hash_string((unsigned char *)"12A5B6cd_"));
+    TEST_ASSERT_EQUAL_INT(5381, hash_string((unsigned char *)""));
```

(Maybe you need to adapt for line width ;-))